libs/ledger-live-common/src/market/api/api.ts

```diff
@@ -241,9 +233,43 @@ async function currencyChartData({
   return { [range]: data.prices };
 }
 
+export async function fetchCurrencyData({
+  counterCurrency,
+  range = "24h",
+  id,
+}: MarketCurrencyChartDataRequestParams): Promise<CurrencyData> {
+  const url =
+    `${ROOT_PATH}/coins/markets?vs_currency=${counterCurrency}` +
+    `&sparkline=true&price_change_percentage=${range}` +
+    `&page=1&&ids=${id}`;
+
+  const { data } = await network({
+    method: "GET",
+    url,
+  });
+
+  /**
+   * @data returns array of one element
+   */
+  return format(data[0], range, cryptoCurrenciesList);
```

Comment:
We shouldn't have any transformations/selectors in fetch functions, rather pass a `selector` into the `useQuery()` this way you aren't running this potentially unnecessary function given the output didn't change.